Handle Ctrl-C gracefully

kwiberg opened the pull request:
Basically, there are two things we need to do: exiting _gracefully_
without emitting backtraces, and exiting _promptly_ without waiting
for tests currently in progress to finish (or even worse, tests that
haven't even started yet).

kwiberg commented:
This is a totally rewritten version that works properly without being dreadfully ugly.

pbos commented on gtest-parallel:
```diff
@@ -18,13 +18,62 @@ import gzip
+    with self.__lock:
+      if self.__got_sigint:
+        p.terminate()
+        raise self.ProcessWasInterrupted
```
Remove this line (and wait for the zombie), we'll still raise this on line 73.

pbos replied:
Maybe add a comment instead of this line though, waiting on the terminated process should be fine.

pbos commented on gtest-parallel:
```diff
@@ -18,13 +18,62 @@ import gzip
+  def got_sigint(self):
+    with self.__lock:
+      return self.__got_sigint
+  def wait_for_or_terminate(self, p):
```
Rename wait(self, p)

kwiberg committed:
Make Ctrl+C detection work on Windows too

kwiberg committed:
Handle Ctrl+C gracefully

Basically, there are two things we need to do: exiting *gracefully*
without emitting backtraces, and exiting *promptly* without waiting
for tests currently in progress to finish (or even worse, tests that
haven't even started yet).
